### DISCUSSION TITLE
Configuring Ingress for whitelisted IP on deploy

### DISCUSSION THREAD

**Discussion by niclascoeli** [microsoft/aspire#2265] [Q&A] [answered]
Testing out deploying some applications using Aspire and so far it seem very nice, but an annoying theme is that I'm unable to customize the deployment step at all (azd deploy).

For example, I can't have ingress set to"Limited to container apps environment" as then the web app isn't avaialble at all, but I don't like to have ingress set to "Accept traffic from anywhere" without a IP restriction. As of now, the Ingress is always reset to private when running "azd deploy" and then I manually set it to public and add the IP restrictions.

I'm sure there's some way to configure the deploy behaviour more in detail, but not sure how? Can anyone point me in the general direction?

**Comment by davidfowl** [comment, marked as answer]
https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?tabs=windows#generate-bicep-from-net-aspire-app-model

By editing bicep by hand.

**Reply by niclascoeli** [reply]
Thanks for the help! Not sure how I missed that...

Do you know how this is supposed to work in the future? Should i just consider 'azd infra synth' as a getting started tool and then be manually editing the genereated files, or will this be a tool similar to EF migrations that should be used to drive the infra? Or something inbetween with manual diffs and patches?

**Reply by davidfowl** [reply]
This tool is the escape hatch. We’re still designing a way to properly this data through into the deployment tool.
